lib/helper/User.js

```diff
@@ -39,44 +36,163 @@ class User {
+    oThis.userWalletFactory = userWalletFactory;
+    oThis.proxyFactory = proxyFactory;
     oThis.auxiliaryWeb3 = auxiliaryWeb3;
-
     oThis.abiBinProvider = new AbiBinProvider();
   }
 
+  /** Generates DelayedRecoveryModule::setup() function data. */
+  getDelayedRecoveryModuleSetupData(recoveryOwnerAddress, recoveryControllerAddress, recoveryBlockDelay) {
+    const oThis = this;
+
+    return oThis.auxiliaryWeb3.eth.abi.encodeFunctionCall(
+      {
+        name: 'setup',
+        type: 'function',
+        inputs: [
+          {
+            type: 'address',
+            name: 'recoveryOwnerAddress'
+          },
+          {
+            type: 'address',
+            name: 'recoveryControllerAddress'
+          },
+          {
+            type: 'uint256',
+            name: 'recoveryBlockDelay'
+          }
+        ]
+      },
+      [recoveryOwnerAddress, recoveryControllerAddress, recoveryBlockDelay]
+    );
+  }
+
+  /**
+   * Generates ProxyFactory::createProxy() function data for
+   * DelayedRecoveryModule proxy's creation.
+   */
+  getDelayedRecoveryModuleCreationData(delayedRecoverySetupData) {
+    const oThis = this;
+
+    return oThis.auxiliaryWeb3.eth.abi.encodeFunctionCall(
+      {
+        name: 'createProxy',
+        type: 'function',
+        inputs: [
+          {
+            type: 'address',
+            name: 'delayedRecoveryModuleMasterCopy'
+          },
+          {
+            type: 'bytes',
+            name: 'delayedRecoverySetupData'
+          }
+        ]
+      },
+      [oThis.delayedRecoveryModuleMasterCopy, delayedRecoverySetupData]
+    );
+  }
+
+  getCreateAndAddModulesData(modulesCreationData) {
+    const oThis = this;
+
+    const ModuleDataWrapper = new oThis.auxiliaryWeb3.eth.Contract([
+      {
+        constant: false,
+        inputs: [
+          {
+            name: 'data',
+            type: 'bytes'
+          }
+        ],
+        name: 'setup',
+        type: 'function'
+      }
+    ]);
+
+    // Remove method id (10) and position of data in payload (64)
+    const reducedModulesCreationData = modulesCreationData.reduce(
+      (acc, data) =>
+        acc +
+        ModuleDataWrapper.methods
+          .setup(data)
+          .encodeABI()
+          .substr(74),
+      '0x'
+    );
+
+    return oThis.auxiliaryWeb3.eth.abi.encodeFunctionCall(
+      {
+        name: 'createAndAddModules',
+        type: 'function',
+        inputs: [
+          {
+            type: 'address',
+            name: 'proxyFactory'
+          },
+          {
+            type: 'bytes',
+            name: 'data'
+          }
+        ]
+      },
+      [oThis.proxyFactory, reducedModulesCreationData]
+    );
+  }
+
   /**
    * Generate the executable data for setup method of GnosisSafe contract.
    *
    * @param owners List of owners of the multisig contract for a user.
    * @param threshold Number of required confirmations for a Safe transaction.
```

Comment:
the params are missing from the documentation; can be fixed later

Reply:
Done.